tss-esapi/tests/integration_tests/error_tests/return_code_tests/resource_manager_tpm_tests.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use std::convert::TryFrom;
+use tss_esapi::{
+    constants::tss::{TPM2_RC_ASYMMETRIC, TPM2_RC_SEQUENCE, TSS2_RESMGR_TPM_RC_LAYER},
+    error::{ReturnCode, TpmResponseCode},
+};
+
+#[test]
+fn test_valid_tpm_resmgr_format_zero_response_code() {
+    let expected_tss_rc = TSS2_RESMGR_TPM_RC_LAYER | TPM2_RC_SEQUENCE;
+    let actual_rc = ReturnCode::try_from(expected_tss_rc)
+        .expect("Failed to convert TPM2_RC_SEQUENCE in the TPM RESMGR layer to a ReturnCode.");
+
+    if let ReturnCode::TpmResourceManager(actual_tpm_response_code) = actual_rc {
+        match actual_tpm_response_code {
+            TpmResponseCode::FormatZero(_) => {}
+            _ => {
+                panic!("TPM2_RC_SEQUENCE in the TPM RESMGR layer did not convert into the expected TpmResponseCode");
+            }
+        }
+    } else {
+        panic!("The TPM RESMGR layer did not convert into the expected ReturnCode");
+    }
+
+    assert_eq!(
+        expected_tss_rc,
+        actual_rc.into(),
+        "ReturnCode::TpmResourceManager did not convert into the expected TSS2_RC value"
+    );
+}
+
+#[test]
+fn test_valid_tpm_resmgr_format_one_response_code() {
+    let expected_tss_rc = TSS2_RESMGR_TPM_RC_LAYER | TPM2_RC_ASYMMETRIC;
+    let actual_rc = ReturnCode::try_from(expected_tss_rc)
+        .expect("Failed to convert TPM2_RC_ASYMMETRIC in the TPM RESMGR layer to a ReturnCode.");
+
+    if let ReturnCode::TpmResourceManager(actual_tpm_response_code) = actual_rc {
+        match actual_tpm_response_code {
+            TpmResponseCode::FormatOne(_) => {}
+            _ => {
+                panic!("TPM2_RC_ASYMMETRIC in the TPM RESMGR layer did not convert into the expected TpmResponseCode");
```

Comment:
Could that be expressed using [`assert_matches`](https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html)?

Reply:
Ugh, I got bitten by the nightly thing again. For stable I meant assert + matches: https://doc.rust-lang.org/std/macro.matches.html

Reply:
In 1.65 I get "Unstable library feature" if add an ```assert_matches!```.

Reply:
Yep. Sorry for that I always forget it's not stabilized yet. Could you check out manual assert + matches macro? (See my second link)